File: 1018. Binary Prefix Divisible By 5/solution.js
/**
 * @param {number[]} nums
 * @return {boolean[]}
 */
 var prefixesDivBy5 = function(nums) {
  let current = 0;
  let result = [];
  
  for (let num of nums){
    current = (current * 2 + num) % 5;
    result.push(current === 0);
  }

  return result;
};

File: 1018. Binary Prefix Divisible By 5/solution.cpp
class Solution {
public:
  vector<bool> prefixesDivBy5(vector<int>& nums) {
    int current = 0;
    vector<bool> result;

    for(int i = 0; i < nums.size(); i++){
      current = (current * 2 + nums[i]) % 5;
      result.push_back(current == 0);
    }

    return result;
  }
};